AirSkyBoat/AirSkyBoat
# Add Dyna Tav Bosses and CSs

**_I affirm:_**


## Please enter a player-facing description

Adds Diabolos Bosses and CSs to Dyna Tav
Adds the inital CoP dyna CS to touching the heiro to other CoP dyna zones (1 CS, repeated across all 3, can see it once)

## Known ToDos
- The Umbral Diabolos are supposed to be NPCs - not mobs.  They function as they did in era - but are not NPCs.
- Technically, when multiple diabolos are spawned - they should have their max HP reduced.  This supposedly works as a cap they cannot regen or rest over - so they will show as 80%, 60%, 40% (iirc).

## What does this pull request do? (Please be technical)

Implements Diabolos bosses including unique TP moves.
Implements CS touching Heiroglyphics.
Implements Win CS (with 4 variations based on mission status)

Some more captures are required for all animation IDs and perhaps some potential scripted behavior of the mobs.
Additionally - more research is required for Club & Shards.
Some era forum posts say Shard is summoned twice, uses an AoE, and despawns each time.  That is what I have went with.

## Steps to test these changes
Touch the heiros for bubu/valk/quf to see starter CS (with enter reqs met or GM)
Touch the heiros for Tav to see the tav starter CS (with enter reqs met or GM)
Touch the heiros after beating Tav - get slightly slighty different CSs if you have beaten AN, beaten cop and zm, beaten zm, or beaten nothing

Run Dyna Tav multiple times killing diaboloses
To just mess with bosses and not care about spawning mechanics/win mechanis/wipe mechanics the following commands are useful
```
!dynaspawnnm TAVNAZIA 110
!dynaspawnnm TAVNAZIA 111
!dynaspawnnm TAVNAZIA 112
!dynaspawnnm TAVNAZIA 113
!adddynatime TAVNAZIA 60
```

## Expected Behavior
DynaBums has a pretty good chartof who can use what.
https://www.dynamisbums.com/strategy/tav.html
Some JP wikis go a bit more in depth - i.e. the dispel the chart on dyna bums is actually death
For more sources - see previous PRs
```
**Shared Traits:**
Resists Stun Highly
No Lullaby
No Sleep
No Silence

**Diabolos NMs have distinguishing characteristics:**
Diabolos Club - has Diabolos's Shard pet.  Shard has AoE dmg move.  Its an instant summon, use dmg move, then despawn.  Twice.
Diabolos Diamond - does not melee, follows Nightmare with Daydream - up to 5 targets.
Diabolos Heart - spams Drain at low life
Diabolos Spade - does not cast magic. Higher def. Maybe spams Nether Blast - but that is not documented well

**Spawning and winning/losing**
When an umbral is aggro'd it despawns and spawns a random Boss Diabolos - and music changes
Umbrals aggro sight only, true sight
When all spawned diabolos are killed - its a win, and all umbrals will despawn.
Diabolos superlink
If players wipe to the diabolos, the last one to disengage will say something and all diabolos and umbrals despawn
```

## Special Deployment Considerations
None
